### DISCUSSION TITLE
How to debug an graphics freezes on Linux guest

### DISCUSSION THREAD

**Discussion by stuaxo** [utmapp/UTM#5855] [Linux]
On a Mac M2, I'm using UTM with Qemu and Ubuntu 22.04 aarch 64 as the guest.

For development I sometimes run our unit tests, which use quite a few resources and often the guest freezes (though the mouse still moves).

I thought it might be virtio fsd, but I disabled all file sharing so it's not that.

How can I start to work out what's going on ?

**Comment by stuaxo** [comment]
OK, so I got the qemu console configured on the built in terminal, so I guess that's a start - and I can do `system-reset` at least and reset the machine (still not ideal since I am running an IDE in it)

**Comment by nielstanis** [comment]
I'm experiencing the same on my M2 running same Ubuntu version; something that showed up over the last couple of weeks.. Curious if anyone can help out with figuring out the problem and possibly fix it..

**Reply by stuaxo** [reply]
Looks very similar to this issue https://github.com/utmapp/UTM/issues/5886

**Reply by stuaxo** [reply]
Running UTM in the terminal shows the spice logs:
```sh
/Applications/UTM.app/Contents/MacOS/UTM
```

There is nothing unusual though - it does say a bunch of things about audio muting / not muting.   As a precaution, in the guest I tried setting all the audio options to muted in GNOME... I don't yet, but it seems to freeze a little less (on the other hand it may just still be a coincidence).

**Reply by stuaxo** [reply]
I added a bridged network and am able to ssh in when it's frozen.

It looks like Java2D (as part of Pycharm) might be outputting something that causes the issue.

**Reply by stuaxo** [reply]
See if disabling your sound device helps - first I muted sound inside my VM, but then I actually edited the settings of the VM and deleted the Sound device.
It seems *much* more stable for me.

I actually *gained* sound functionality, because I installed pulseaudio on my MacOS host using homebrew and routed audio to it via tcpip - it's quick enough for things like youtube to work acceptably.

**Comment by stuaxo** [comment]
With upgrades I still have the issue:
- Ubuntu to 23.04, on current 6.6.4 kernel from mainline.
- Mesa to 24.0 from oibaf ppa

Would attach a debug.log but it's a bit on the large side.

**Comment by stuaxo** [comment]
@nielstanis closing, as I'm no longer using UTM+Qemu so can't give feedback, and most of this chat is mostly spam from me :) feel free to open a new one if it's still an issue.
